### DISCUSSION TITLE
Change a part of a 3D model to a new part which is loaded from a local or remote path(eg. change short hair to long hair)

### DISCUSSION THREAD

**Discussion by whereisnothing** [google/filament#5920] [General]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.

**Describe the solution you'd like**
First we show a 3D glb model which is composed of many parts.Then take hair for example,we now load a new part(ie. hair) from a local path on Android phone or a remote path on server which contains a changed part of the model.At last we change the original part (ie. hair) with the new hair loaded from the new path.How can we achieve this?Thanks very much.

**Describe alternatives you've considered**

**OS and backend**
Is this feature OS/backend specific? If so, explain why.

**Comment by pixelflinger** [comment]
why can't you do this manually?

**Comment by prideout** [comment]
Yes, like @pixelflinger said, this could be done manually by simply removing the old hair from the Scene and adding the new hair.  gltfio exposes all the underlying Filament primitives, so anything is possible.  It gets a little trickier if the hair is part of a skinning system, but gltfio provides attachSkin / detachSkin methods to make this easier.

Alternatively, you could split up your original glb, so that the hair is separate from the other parts of the model.

**Reply by whereisnothing** [reply]
What I want to do: I want to replace a part of a 3D glb model with a new part loaded from a glb file which contains only a part of a 3D glb model.
I tried the following,but it can't succeed(The new part malposition and can't blend into the original 3D glb model).Please help me out,thanks.
1. Load a new 3D glb model part from a file.Load asset from buffer: FilamentAsset createAssetFromBinary(@NonNull Buffer buffer)
2. Pop renderables from:FilamentAsset.popRenderables(renderables)
3. Reparent new instantiated entity and add new entity to the scene:     
```kotlin           
                rcm.setLayerMask(entity, 0xff, 0xff)
                tm.setParent(entity, oldEntityParent)
                scene.addEntity(entity)
```

5. Remove old entity from scene

**Reply by whereisnothing** [reply]
Could you please give me a demo on how to replace a part of a 3D glb model?I tried many apis,but can't succeed(What I want to achieve is as follows:Replace the original hair of a 3D glb model with a new hair.  But the result is: The new hair malposition and I can't successfully use attachSkin api to blend the new hair into the skinning system of the original 3D glb model).You can see the video I uploaded to know what I'm working now.Hope your reply.Thanks for your time.



https://user-images.githubusercontent.com/6499374/186085329-952bec4d-e09f-4bef-b00f-5058b816a646.mp4

**Reply by dandingol03** [reply]
> Could you please give me a demo on how to replace a part of a 3D glb model?I tried many apis,but can't succeed(What I want to achieve is as follows:Replace the original hair of a 3D glb model with a new hair. But the result is: The new hair malposition and I can't successfully use attachSkin api to blend the new hair into the skinning system of the original 3D glb model).You can see the video I uploaded to know what I'm working now.Hope your reply.Thanks for your time.
> 
>  SVID_20220822_165008_part.mp4

HI, have you fixed this problem. I also encounter the same problem
